cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceTaskContextProvider.java

```diff
@@ -111,13 +113,24 @@ protected void shutDown() throws Exception {
    * Returns the {@link BasicMapReduceTaskContext} for the given task.
```

Comment:
unrelated to this PR, but I just re-read this code... would it make sense to invalidate the cache on shutdown? To make sure all contexts can be gc'ed even if this provider is not gc'ed immediately?